Core/src/com/infiniteautomation/mango/spring/components/PasswordResetService.java

```diff
@@ -203,4 +203,17 @@ private void sendEmail(User user, String token) {
             log.error("Couldn't send password reset email", e);
         }
     }
+
+    public User systemSetup(String password, Map<String, Object> settings) {
+        PermissionHolder holder = Common.getUser();
+        User user = holder.getUser();
+
+        // don't want to change the passed in user in case it comes from the cache (in which case another thread might use it)
+        User copy = usersService.get(user.getId());
+        copy.setPlainTextPassword(password);
+        usersService.ensureValid(copy, holder);
+
+        SystemSettingsDao.instance.updateSettings(settings);
```

Comment:
Can the DAO be injected in the constructor? We should avoid using the static `instance` field.